hazelcast/src/main/java/com/hazelcast/instance/HazelcastProperties.java

```diff
@@ -103,7 +109,12 @@ public boolean getBoolean(HazelcastProperty groupProperty) {
      * @throws NumberFormatException if the value cannot be parsed
      */
     public int getInteger(HazelcastProperty groupProperty) {
-        return Integer.parseInt(properties[groupProperty.getIndex()]);
+        int configuredValue = Integer.parseInt(properties[groupProperty.getIndex()]);
+        if (configuredValue < 0) {
```

Comment:
The ones with default value `-1` has different meanings. 
For example we have
"hazelcast.operation.thread.count", "hazelcast.operation.generic.thread.count"
in which `-1` use a default value that hazelcast see fits(mostly dynamically calculated)

We have `-1` where it means feature is disabled.

I could add a check like following
```
if(configuredValue == -1 && defaultValue == -1){
     return -1;
}
```
So that when user specifically writes `-1`  that which have `-1`  as default value, we will not print anything to user.
I have found an exception that does not fit to my suggestion
"hazelcast.query.max.local.partition.limit.for.precheck" default value is 3 and can be disabled by -1

Then, adding a field to GroupProperty to differentiate `-1` means `disable` the feature.
@Donnerbart Do you think is that OK ?